src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -100,13 +100,14 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
             var isKey = Transformer.ValueColumnType.IsKey;
             var columnType = (isKey) ? PrimitiveType.FromKind(DataKind.U4) :
                                     Transformer.ValueColumnType;
+            var metadata = SchemaShape.Create(Transformer.ValueColumnMetadata.Schema);
             foreach (var (Input, Output) in _columns)
             {
                 if (!inputSchema.TryFindColumn(Input, out var originalColumn))
                     throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", Input);
 
                 // Get the type from TOutputType
```

Comment:
What is TOutputType here? Is it not Output?

Reply:
That is old, I used to have the generic arguments be TInputType, TOutputType, but then changed to TKeyType and TValueType. I will update the comment.

---
In reply to: [247282739](https://github.com/dotnet/machinelearning/pull/2098#discussion_r247282739) [](ancestors = 247282739)